### PR TITLE
Change `CMakePresets.json` sample files to use schema version 1

### PR DESCRIPTION
**Pull request by vicroms**
To make our `CMakePresets.json` sample files we need to make some changes:

* `version` should be 1
* `binaryDir` and `generator` are required fields
* `toolchainFile` is not allowed, instead use `cacheVariables` and set `CMAKE_TOOLCHAIN_FILE`

### CODE REVIEW

**Review by AugP — APPROVED** [review]

**Review by JavierMatosD — APPROVED** [review]

File: vcpkg/get_started/get-started-vscode.md [hunk 1]
```diff
@@ -158,20 +158,21 @@ you'll need to provide the `vcpkg.cmake` toolchain file. To automate this,
 create a `CMakePresets.json` file in the "helloworld" directory with the
 following content:
 
+:::code language="cmake" source="../examples/snippets/get-started/CMakePresets.json":::
```
Comment (vicroms) [inline vcpkg/get_started/get-started-vscode.md:161]:
```suggestion
```

File: vcpkg/get_started/get-started-vscode.md [hunk 2]
```diff
@@ -158,20 +158,20 @@ you'll need to provide the `vcpkg.cmake` toolchain file. To automate this,
...
-            "name": "vcpkg",
-            "cacheVariables": {
-                "CMAKE_TOOLCHAIN_FILE": {
-                   "value": "$env{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake",
-                    "type": "FILEPATH"
-                }
-            }
-        }
-    ]
+  "version": 1,
```
Comment (data-queue) [inline vcpkg/get_started/get-started-vscode.md:164]: The VS Code CMake Tools extension only supports version 2 or later for `CMakePresets.json`.

File: vcpkg/examples/snippets/get-started/CMakePresets.json [hunk 3]
```diff
@@ -1,12 +1,13 @@
...
-            "name": "default",
-            "binaryDir": "${sourceDir}/build",
-            "cacheVariables": {
-                "CMAKE_TOOLCHAIN_FILE": "$env{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake"
-            }
-        }
-    ]
-}
\ No newline at end of file
+  "version": 1,
```
Comment (vicroms) [inline vcpkg/examples/snippets/get-started/CMakePresets.json:2]:
```suggestion
  "version": 2,
```

File: vcpkg/get_started/get-started-vs.md [hunk 4]
```diff
@@ -127,15 +127,17 @@ This tutorial shows you how to create a C++ "Hello World" program that uses the
   
     ```json
     {
-    "version": 3,
-    "configurePresets": [
+      "version": 1,
```
Comment (vicroms) [inline vcpkg/get_started/get-started-vs.md:130]:
```suggestion
      "version": 2,
```

File: vcpkg/get_started/get-started-vscode.md (hunk 2 above)
Comment (vicroms) [inline vcpkg/get_started/get-started-vscode.md:164]:
```suggestion
  "version": 2,
```

File: vcpkg/get_started/get-started-vscode.md [hunk 5]
```diff
@@ -180,16 +180,16 @@ following content:
...
-    "configurePresets": [
-        {
-            "name": "default",
-            "inherits": "vcpkg",
-            "environment": {
-                "VCPKG_ROOT": "<path to vcpkg>"
-            }
-        }
-    ]
+  "version": 1,
```
Comment (vicroms) [inline vcpkg/get_started/get-started-vscode.md:183]:
```suggestion
  "version": 2,
```

File: vcpkg/users/buildsystems/snippets/cmake-integration/CMakePresets.json [hunk 6]
```diff
@@ -1,9 +1,11 @@
 {
-    "version": 3,
-    "configurePresets": [
-        {
-            "name": "debug",
-            "toolchainFile": "$env{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake"
-        }
-    ]
+  "version": 1,
```
Comment (vicroms) [inline vcpkg/users/buildsystems/snippets/cmake-integration/CMakePresets.json:2]:
```suggestion
  "version": 2,

```

File: vcpkg/users/buildsystems/snippets/cmake-integration/CMakeUserPresets.json [hunk 7]
```diff
@@ -1,9 +1,12 @@
 {
-    "version": 3,
-    "configurePresets": [
-        {
-            "name": "debug",
-            "toolchainFile": "C:/src/vcpkg/scripts/buildsystems/vcpkg.cmake"
-        }
-    ]
+  "version": 1,
```
Comment (vicroms) [inline vcpkg/users/buildsystems/snippets/cmake-integration/CMakeUserPresets.json:2]:
```suggestion
  "version": 2,

```

**Pull request merged**